### PR TITLE
roon-server: init 100500363

### PR DESCRIPTION
**Pull request by lovek323**
###### Motivation for this change

I use this to organise and listen to my music.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I haven't played with modules much, so I'd appreciate any suggestions/feedback on how this could improve. There is one thing I need specific help with - I'll add a comment inline.

### CODE REVIEW

File: pkgs/servers/roon/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      $out/opt/RoonMono/bin/mono-sgen
+
+    # Checkers
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${alsaLib}/lib" \
+      $out/opt/Appliance/check_alsa
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      $out/opt/Appliance/check_bincompat
```
Comment (lovek323) [inline pkgs/servers/roon/default.nix]: Is there a better way of running the `patchelf` here?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -13807,6 +13807,8 @@ with pkgs;
     boost = boost159;
   };
 
+  roon = callPackage ../servers/roon { };
```
Comment (lovek323) [inline pkgs/top-level/all-packages.nix]: Should this be called `roon-server` to be in line with the nixos module, or is it fine to leave it at `roon`?

Comment (lovek323) [reply pkgs/top-level/all-packages.nix]: I'll update it to be `roon-server`, because "roon" would be the UI instead of the server.

**Comment by reanimus** [comment]
I'd also be interested in seeing this merged, as I'd love to move my media server to NixOS :)

File: nixos/modules/misc/ids.nix [hunk 3]
```diff
@@ -338,6 +338,7 @@
       minetest = 311;
       rss2email = 312;
       cockroachdb = 313;
+      roon-server = 314;
```
Comment (Infinisil) [inline nixos/modules/misc/ids.nix]: No need to allocate a static uid,gid

File: nixos/modules/services/audio/roon-server.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  config = mkIf cfg.enable {
+    systemd.services.roon-server = {
+      after = [ "network.target" ];
+      description = "Roon Server";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Environment = "\"ROON_DATAROOT=${cfg.dataDir}\"";
+        ExecStart = "${pkgs.bash}/bin/bash ${pkgs.roon-server}/opt/start.sh";
+        ExecStartPre = "${pkgs.coreutils}/bin/mkdir -p ${cfg.dataDir} && chown ${cfg.user}:${cfg.group} ${cfg.dataDir}";
```
Comment (Infinisil) [inline nixos/modules/services/audio/roon-server.nix]: Use `systemd.tmpfiles` instead

File: nixos/modules/services/audio/roon-server.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        LimitNOFILE = 8192;
+        PermissionsStartOnly = true;
+        User = "${cfg.user}";
```
Comment (Infinisil) [inline nixos/modules/services/audio/roon-server.nix]: Just `User = cfg.user;` works. And you should also set `Group =`

File: nixos/modules/services/audio/roon-server.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+  config = mkIf cfg.enable {
+    systemd.services.roon-server = {
+      after = [ "network.target" ];
+      description = "Roon Server";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Environment = "\"ROON_DATAROOT=${cfg.dataDir}\"";
+        ExecStart = "${pkgs.bash}/bin/bash ${pkgs.roon-server}/opt/start.sh";
```
Comment (Infinisil) [inline nixos/modules/services/audio/roon-server.nix]: Why not call the script directly?

File: pkgs/servers/roon-server/default.nix [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+stdenv.mkDerivation rec {
+  name = "roon-server";
+  version = "100500363";
+
+  src = fetchurl {
+    url = "http://download.roonlabs.com/updates/stable/RoonServer_linuxx64_${version}.tar.bz2";
+    sha256 = "1pdlglhmsm0l4k6g4l97ckw596ckw8nnxii60j1xg994kdiikz3s";
+  };
+
+  propagatedBuildInputs = [ alsaLib alsaUtils cifs_utils ffmpeg libav mono ];
```
Comment (Infinisil) [inline pkgs/servers/roon-server/default.nix]: Pretty sure you can just remove this line

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by peterhoeg** [comment]
Closing in favour of #63551

**Pull request closed without merging**